src/rgw/setup/rgw.py

```diff
@@ -187,36 +187,78 @@ def cleanup(conf: MappedConf, pre_factory: bool = False):
         return 0
 
     @staticmethod
-    def upgrade(conf: MappedConf):
+    def _update_rgw_property(conf: MappedConf, config_index:str, config_key: str, config_val: str):
+        """Add/Update key in rgw config file."""
```

Comment:
this comment is misleading
code is actually changing value of rgw_config, for which GConf config_key is passed

We can have something like
> Update RGW Config based on changed Gconf values

And how this code is adding new config keys exactly?

Reply:
updated the description.
In const.SVC_CONFIG_DICT mapping , local gconf mapping key with gconf key passed in changeset file with our 
if both are same then only we are updating its corresponding rgw config property.

Reply:
alright. for adding new key also i understand the logic now